File: LibraryManagementSystem/Backend/LibraryManagementSystem/LibraryManagementSystem.Entities/Dtos/Security/UserDto.cs
namespace LibraryManagementSystem.Entities.Dtos.Security
{
    public class UserDto
    {
        public int UserId { get; set; }
        public string? UserName { get; set; }
        public string? Email { get; set; }
        public bool Active { get; set; }
        public string? AccessToken { get; set; }
        public string? RefreshToken { get; set; }
        public DateTime RefreshTokenExpiryTime { get; set; }
        public bool LockoutEnabled { get; set; }
        public int AccessFailedCount { get; set; }
        public List<RoleDto>? Roles { get; set; } = new();
    }
}

File: LibraryManagementSystem/Frontend/library-management-system-web/src/app/entities/dtos/security/user-dto.ts
import { RoleDto } from "./role-dto";

export interface UserDto {
    userId: number;
    userName: string | null;
    email: string | null;
    active: boolean;
    accessToken: string | null;
    refreshToken: string | null;
    refreshTokenExpiryTime: Date;
    lockoutEnabled: boolean;
    accessFailedCount: number;
    roles: RoleDto[] | null;
}